source/extensions/omni.isaac.lab/omni/isaac/lab/sim/spawners/sensors/sensors_cfg.py

```diff
@@ -71,6 +73,72 @@ class PinholeCameraCfg(SpawnerCfg):
     the camera output on the GUI and not accidentally moving the camera through the GUI interactions.
     """
 
+    @classmethod
+    def from_intrinsic_matrix(
+        cls,
+        intrinsic_matrix: list[float],
+        width: int,
+        height: int,
+        projection_type: str = "pinhole",
+        clipping_range: tuple[float, float] = (0.01, 1e6),
+        focal_length: float = 24.0,
+        focus_distance: float = 400.0,
+        f_stop: float = 0.0,
+        horizontal_aperture: float = 20.955,
+        horizontal_aperture_offset: float = 0.0,
+        vertical_aperture_offset: float = 0.0,
+        lock_camera: bool = True,
+    ) -> PinholeCameraCfg:
+        """Create a PinholeCameraCfg from an intrinsic matrix. The intrinsic matrix is a 3x3 matrix that
+        defines the mapping between the 3D world coordinates and the 2D image. The matrix is defined as:
+
+        .. math::
+            \\begin{bmatrix}
+            f_x & 0 & c_x \\\\
+            0 & f_y & c_y \\\\
+            0 & 0 & 1
+            \\end{bmatrix}
+
+        Args:
+            intrinsic_matrix: Intrinsic matrix of the camera (shape: (9,)).
+            width: Width of the image (in pixels).
+            height: Height of the image (in pixels).
+            projection_type: Type of projection to use for the camera. Defaults to "pinhole".
+            clipping_range: Near and far clipping distances (in m). Defaults to (0.01, 1e6).
+            focal_length: Perspective focal length (in cm). Defaults to 24.0cm.
+            focus_distance: Distance from the camera to the focus plane (in m). Defaults to 400.0.
+            f_stop: Lens aperture. Defaults to 0.0, which turns off focusing.
+            horizontal_aperture: Horizontal aperture (in mm). Defaults to 20.955mm.
+            horizontal_aperture_offset: Offsets Resolution horizontally. Defaults to 0.0.
+            vertical_aperture_offset: Offsets Resolution vertically. Defaults to 0.0.
+            lock_camera: Locks the camera in the Omniverse viewport. Defaults to True.
+
+        Returns:
+            PinholeCameraCfg: The configuration for the pinhole camera pattern.
+        """
+        # extract parameters from matrix
+        f_x = intrinsic_matrix[0]
+        c_x = intrinsic_matrix[2]
+        f_y = intrinsic_matrix[4]
+        c_y = intrinsic_matrix[5]
+        # resolve parameters for usd camera
+        horizontal_aperture = width * focal_length / f_x
+        # vertical_aperture = height * focal_length / f_y
```

Comment:
Maybe we should make a note here as a TODO to check vertical aperture and offset?

Otherwise for now throw a warning that these parameters don't do anything?

Reply:
are you sure that they don't do anything?